packages/cli/src/TestWebhooks.ts

```diff
@@ -135,15 +136,34 @@ export class TestWebhooks implements IWebhookManager {
 				}
 			} catch {}
 
-			// Delete webhook also if an error is thrown
-			if (timeout) clearTimeout(timeout);
+			/**
+			 * Multi-main setup: In a manual webhook execution, the main process that
+			 * handles a webhook might not be the same as the main process that created
+			 * the webhook. If so, after the test webhook has been successfully executed,
+			 * the handler process commands the creator process to clear its test webhooks.
+			 */
+			if (
+				this.multiMainSetup.isEnabled &&
+				sessionId &&
+				!this.push.getBackend().hasSessionId(sessionId)
+			) {
+				const payload = { webhookKey: key, workflowEntity, sessionId };
+				void this.multiMainSetup.publish('clear-test-webhooks', payload);
+				return;
+			}
 
-			await this.registrations.deregisterAll();
+			this.clearTimeout(key);
```

Comment:
Thank you, please see [`302103e`](https://github.com/n8n-io/n8n/pull/8267/commits/302103eeb2629a4b0247109d406013d108e048cc).

Reply:
Re: ensuring user cancellation triggers a push to the UI if no registration is found - the session ID to push to is in the test webhook registrations, of which there are none in this scenario, so we'd need to refactor to source the session ID from elsewhere, but in `/test-webhook/:id` the session ID is not available. 

As discussed, since we already have the safeguard + buffer in place mostly preventing a user cancellation where there are no registrations, let's move forward and revisit later if needed.